### ISSUE TITLE
please declare copy spec as input

### ISSUE THREAD

**Issue by Vampire** [wooga/atlas-github#30]
If you execute `githubPublish` task it does not implicitly depend on the artifact producing tasks, though the tasks are used as copy sources.
Usually if you use a task as `from` and so its outputs, it is automatically added as task dependency too.
In case of the GithubPublish task this is not the case and you have to manually define the dependeny.
I think you just have to declare them as inputs in the task and it should work automagically.

**Comment by Larusso** [comment]
I think I need to check for the ‘Buildable’ interface and add its build dependencies.

**Comment by Vampire** [comment]
No, as I said, you just have to declare your inputs.
This change to 1.3.0 was enough to get the implicit task dependencies:
```diff
diff --git a/src/main/groovy/wooga/gradle/github/publish/tasks/GithubPublish.groovy b/src/main/groovy/wooga/gradle/github/publish/tasks/GithubPublish.groovy
index d53ae4d..75af4e9 100644
--- a/src/main/groovy/wooga/gradle/github/publish/tasks/GithubPublish.groovy
+++ b/src/main/groovy/wooga/gradle/github/publish/tasks/GithubPublish.groovy
@@ -307,6 +307,7 @@ class GithubPublish extends AbstractGithubTask implements GithubPublishSpec {
      */
     @Override
     GithubPublish from(Object... sourcePaths) {
+        inputs.files(sourcePaths)
         assetsCopySpec.from(sourcePaths)
         processAssets = true
         this
@@ -334,6 +335,7 @@ class GithubPublish extends AbstractGithubTask implements GithubPublishSpec {
      */
     @Override
     GithubPublish from(Object sourcePath, Action<? super CopySpec> configureAction) {
+        inputs.files(sourcePath)
         assetsCopySpec.from(sourcePath, configureAction)
         processAssets = true
         this
```

**Comment by Larusso** [comment]
Ah ok thanks. I already prepared a test.

**Comment by Larusso** [comment]
Fixed in #34 and #35 (for `2.x`)
Next patch release will include these fixes
